libraries/WebServer/examples/MultiHomedServers/MultiHomedServers.ino

```diff
@@ -0,0 +1,173 @@
+ *      http://stationaddress:8080  "hello from server0"
+ *      http://stationaddress:8081  "hello from server1"
+ *      http://stationaddress:8082  browser reports failure to connect
+ *      
+ *  2. when client is connected to the soft AP:
+ *  
+ *      Request URL                 Response
+ *      -----------                 --------
+ *      http://softAPaddress:8080   "hello from server0"
+ *      http://softAPaddress:8082   "hello from server2"
+ *      http://softAPaddress:8081   browser reports failure to connect
+ *
+ *  3. Repeat 1 and 2 above with esp32.local in place of stationaddress and softAPaddress, respectively.
+ *  
+ * MultiHomedServers was originally based on HelloServer.
+ */
+
+const char* ssid = "........";
+const char* password = "........";
+const char *apssid = "ESP32";
+
+WebServer *server0, *server1, *server2;
+
+const int led = 13;
+
+void handleRoot(WebServer *server, const char *content) {
+  digitalWrite(led, 1);
+  server->send(200, "text/plain", content);
+  digitalWrite(led, 0);
+}
+
+void handleRoot0() {
+  handleRoot(server0, "hello from server0");
+}
+
+void handleRoot1() {
+  handleRoot(server1, "hello from server1");
+}
+
+void handleRoot2() {
+  handleRoot(server2, "hello from server2");
+}
+
+void handleNotFound(WebServer *server) {
+  digitalWrite(led, 1);
+  String message = "File Not Found\n\n";
+  message += "URI: ";
+  message += server->uri();
+  message += "\nMethod: ";
+  message += (server->method() == HTTP_GET) ? "GET" : "POST";
+  message += "\nArguments: ";
+  message += server->args();
+  message += "\n";
+  for (uint8_t i = 0; i < server->args(); i++) {
+    message += " " + server->argName(i) + ": " + server->arg(i) + "\n";
+  }
+  server->send(404, "text/plain", message);
+  digitalWrite(led, 0);
+}
+
+void handleNotFound0() {
+  handleNotFound(server0);
+}
+
+void handleNotFound1() {
+  handleNotFound(server1);
+}
+
+void handleNotFound2() {
+  handleNotFound(server2);
+}
+
+void setup(void) {
+  pinMode(led, OUTPUT);
+  digitalWrite(led, 0);
+  Serial.begin(115200);
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(ssid, password);
+  Serial.println("");
+
+  // Wait for connection
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("");
+  Serial.print("Connected to ");
+  Serial.println(ssid);
+  Serial.print("IP address: ");
+  Serial.println(WiFi.localIP());
+  if (!WiFi.softAP(apssid)) {
+    Serial.println("failed to start softAP");
+    for (;;) {
+        digitalWrite(led, 1);
+        delay(100);
+        digitalWrite(led, 0);
+        delay(100);
+    }
+  }
+  Serial.print("Soft AP: ");
+  Serial.print(apssid);
+  Serial.print(" IP address: ");
+  Serial.println(WiFi.softAPIP());
+
+  if (MDNS.begin("esp32")) {
+    Serial.println("MDNS responder started");
+  }
+
+  server0 = new WebServer(8080);
+  server1 = new WebServer(WiFi.localIP(), 8081);
+  server2 = new WebServer(WiFi.softAPIP(), 8082);
```

Comment:
for this test to be valid, both servers should be on the same port. That is how you can make sure that the functionality works.

Reply:
Changed port numbers as suggested. Expected results returned by test script. Pushed new commit. "All checks have passed".